src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -22,6 +22,25 @@
 
 (def clip (partial merge table-style/clip-text))
 
+; ask what other types it could be. Maps?
+(defn get-typed [string-input input-type]
+  (if (or (= input-type "int")
+          (string/includes? input-type "array")
+          (= input-type "string"))
+    ; if conversion to json is unsuccessful (as a result of invalid json), just return string
+    (try (utils/parse-json-string string-input)
+         (catch js/Error e string-input))
```

Comment:
I think what you're after is behaviour looks like this (thing in box -> json value):

`42` -> `42` (ints are json)
`this.foo` -> `"this.foo"` (expressions aren't in quotes and are thus invalid json, so you have to represent them as strings)
`"bees"` -> `"\"bees\""` (things in quotes are valid json, but you have to differentiate them from expressions, so you have to escape the quotes)
`[1,"2"]` ->`[1,"2"]` (arrays are valid json (even with mixed element types))
`[1,"2"` -> `"[1,\"2'\""` (not a valid array, represent as string and escape quotes)

So that's:

* represent json as json where possible
* represent expressions as strings
* string literals end up with escaped quotes (to differentiate from expressions)
* invalid json ends up in a string with escaped quotes

Reply:
In Scala I'd do this (I think):

```
//input is the full context of the text box, including any quotes the user typed
input.parseJson match {
	case Success(JsString(str)) => JsString(input.escapeme())
	case Success(JsValue(other)) => JsValue(other)
	case Failure(_) => JsString(input.escapeme())
}
```

Reply:
I just did some manual testing and `String`, `Int` and `Array[String]` work. Here are examples of types that weren't rendered as I expected they should:
```
{
  "w.t.inFloat": "1.5",
  "w.t.inStringMap": "{\"foo\":\"bar\"}",
  "w.t.inFile": "\"gs://foo/bar\"",
  "w.t.inBoolean": "true"
}
```
All of these have extra outer quotes that shouldn't be there.

Reply:
@helgridly I was under the impression that someone downloading the inputs json would potentially be using it to submit directly to Cromwell, in which case we'd want `"bee"` -> `"bee"` and this._ and workspace._ would be illegal. But, reading the ticket, it says for use in other workspaces, so you're right :)

@breilly2 Yes, I will add those types as well